test/specs/deployAndRetrieve.e2e.ts

```diff
@@ -5,26 +5,331 @@
+    const textAfterRetrieve = await textEditor.getText();
+
+    // Verify Output tab
+    const outputPanelText = await utilities.attemptToFindOutputPanelText(
+      'Salesforce CLI',
+      'Starting SFDX: Retrieve Source from Org',
+      10
+    );
+    utilities.log('Retrieve time - 2: ' + (await utilities.getOperationTime(outputPanelText!)));
+    expect(outputPanelText).not.toBeUndefined();
+    expect(outputPanelText).toContain('Retrieved Source');
+    expect(outputPanelText).toContain(`MyClass    ApexClass  ${pathToClass}.cls`);
+    expect(outputPanelText).toContain(`MyClass    ApexClass  ${pathToClass}.cls-meta.xml`);
+    expect(outputPanelText).toContain('ended SFDX: Retrieve Source from Org');
+    // Retrieve operation will overwrite the file, hence the the comment will remain as before the modification
+    expect(textAfterRetrieve).not.toContain('modified comment');
+  });
+
+  step('Prefer Deploy on Save when `Push or deploy on save` is enabled', async () => {
+    // Clear the Output view first.
+    let outputView = await utilities.openOutputView();
+    await outputView.clearText();
+
+    const workbench = await browser.getWorkbench();
+    await utilities.runCommandFromCommandPrompt(
+      workbench,
+      'Preferences: Open Workspace Settings',
+      5
+    );
+    await browser.keys(['p', 'u', 's', 'h', 'Space', 'o', 'n', 'Space', 's', 'a', 'v', 'e']);
+
+    const pushOrDeployOnSaveBtn = await $(
+      'div[title="salesforcedx-vscode-core.push-or-deploy-on-save.enabled"]'
+    );
+    await pushOrDeployOnSaveBtn.click();
+    await utilities.pause(1);
+
+    const preferDeployOnSaveBtn = await $(
+      'div[title="salesforcedx-vscode-core.push-or-deploy-on-save.preferDeployOnSave"]'
+    );
+    await preferDeployOnSaveBtn.click();
+    await utilities.pause(1);
+
+    // Clear all notifications so clear output button is reachable
+    await utilities.runCommandFromCommandPrompt(
+      workbench,
+      'Notifications: Clear All Notifications',
+      1
+    );
+
+    // Clear the Output view first.
+    outputView = await utilities.openOutputView();
+    await outputView.clearText();
+    // Modify the file and save to trigger deploy
+    const editorView = await workbench.getEditorView();
+    const textEditor = (await editorView.openEditor('MyClass.cls')) as TextEditor;
+    await textEditor.setTextAtLine(2, `\t// let's trigger deploy`);
+    await textEditor.save();
+    await utilities.pause(1);
+    // Wait for the command to execute
+    await utilities.waitForNotificationToGoAway(
+      workbench,
+      'Running SFDX: Deploy Source to Org',
+      utilities.FIVE_MINUTES
+    );
+    // At this point there should be no conflicts since this is a new class.
+    const successNotificationWasFound = await utilities.notificationIsPresent(
+      workbench,
+      'SFDX: Deploy Source to Org successfully ran'
+    );
+    expect(successNotificationWasFound).toBe(true);
```

Comment:
@CristiCanizales Is mac runner failing/ flaky on this expect statement?

Reply:
no! The flaky part on remote is when trying to get the setting